openmetadata-service/src/main/java/org/openmetadata/service/search/elasticsearch/ElasticSearchClient.java

```diff
@@ -691,6 +691,72 @@ public SearchResultListMapper listWithOffset(
     }
   }
 
+  @Override
+  public SearchResultListMapper listWithDeepPagination(
```

Comment:
Do you know if we can use `listWithOffset` from the searchRepository?

Reply:
The issue with listWithOffset is that it limits to 10.000 results:
> By default, you cannot use from and size to page through more than 10,000 hits. This limit is a safeguard set by the [index.max_result_window](https://www.elastic.co/guide/en/elasticsearch/reference/current/index-modules.html#index-max-result-window) index setting. If you need to page through more than 10,000 hits, use the [search_after](https://www.elastic.co/guide/en/elasticsearch/reference/current/paginate-search-results.html#search-after) parameter instead.

Reply:
Make sense. Should we change it in `listWithOffset` to use search_after, I think other calls using the same endpoint may end up facing the same issue with the 10K limit --